### PR TITLE
Add per-frame operator

### PR DESCRIPTION
**Pull request by stiepan**
## Category:

**New feature** (*non-breaking change which adds functionality*) 

## Description:

* Adds new operator: per-frame that sets the input tensor a sequence-like layout. It is meant to be used with sequence processing operators to handle per-frame arguments.
* The operators description presents brief example on specifying per-frame argument.
* Modifies docs generation, so that the documentation of per-frame supporting arguments links to the per_frame operator.

### Key points relevant for the review:
The operator is CPU only for the same reason the fn.transofrms are - it is supposed to be used only on arguments to other operators.

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [x] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [x] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: PERFR.01-PERFR.05

**JIRA TASK**: DALI-2566

### CODE REVIEW

**Comment by stiepan** [comment]
!build

File: dali/pipeline/operator/op_schema.h [hunk 1]
```diff
@@ -291,6 +291,14 @@ graph even if its outputs are not used.)code", false);
     return *this;
   }
 
+  /**
+   * @brief Notes that operator's tensor arguments can be specified per-frame
+   */
+  DLL_PUBLIC inline OpSchema& SupportPerFrameArgs() {
+    support_per_frame_ = true;
+    return *this;
+  }
```
Comment (jantonguirao) [inline dali/pipeline/operator/op_schema.h:300]: I think what we might need fine grain control of which exact argument makes sense to be provided per frame.

Comment (mzient) [reply dali/pipeline/operator/op_schema.h:300]: This should be inferred from argument descriptions.

**Review by jantonguirao — APPROVED** [review]

**Review by prak-nv — APPROVED** [review]

File: dali/python/nvidia/dali/ops.py [hunk 2]
```diff
@@ -130,6 +130,8 @@ def _get_kwargs(schema):
                     default_value = ast.literal_eval(default_value_string)
                     type_name += ", default = `{}`".format(_default_converter(dtype, default_value))
             doc += schema.GetArgumentDox(arg)
+            if schema.ArgSupportsPerFrameInput(arg):
```
Comment (mzient) [inline dali/python/nvidia/dali/ops.py:133]: This requires the other PR to work, doesn't it?

Comment (stiepan) [reply dali/python/nvidia/dali/ops.py:133]: Yeah, it needs to be rebased.

Comment (stiepan) [reply dali/python/nvidia/dali/ops.py:133]: Just wanted to have it ready here so that it could be read, reviewed and discussed if needed.

**Comment by stiepan** [comment]
!build

**Comment by stiepan** [comment]
!build

**Pull request merged**